packit_service/service/api/copr_builds.py

```diff
@@ -41,22 +41,50 @@ class CoprBuildsList(Resource):
     @ns.expect(pagination_arguments)
     @ns.response(HTTPStatus.PARTIAL_CONTENT, "Copr builds list follows")
     def get(self):
-        """ List all Copr builds. From 'copr-builds' hash, filled by service. """
-        # I know it's expensive to first convert whole dict to a list and then slice the list,
-        # but how else would you slice a dict, huh?
-        result = []
-        for k, cb in CoprBuild.db().get_all().items():
-            if k == LAST_PK:
-                # made-up keys, remove LAST_PK
-                continue
-            cb.pop("identifier", None)  # see PR#179
-            result.append(cb)
-
-        first, last = indices()
-        resp = make_response(dumps(result[first:last]), HTTPStatus.PARTIAL_CONTENT)
-        resp.headers["Content-Range"] = f"copr-builds {first + 1}-{last}/{len(result)}"
-        resp.headers["Content-Type"] = "application/json"
-        return resp
+        """ List all Copr builds. """
+
+        # Return relevant info thats concise
+        # Usecases like the packit-dashboard copr-builds table
+
+        with get_sa_session() as session:
+            builds_list = session.query(CoprBuild).all()
+            result = []
+            checklist = []
+            for build in builds_list:
+                if int(build.build_id) not in checklist:
+                    build_dict = {
+                        "project": build.project_name,
+                        "owner": build.owner,
+                        "repo_name": build.pr.project.repo_name,
+                        "build_id": build.build_id,
+                        "status": build.status,
+                        "chroots": [],
+                        "build_submitted_time": build.build_submitted_time.strftime(
+                            "%d/%m/%Y %H:%M:%S"
+                        ),
+                        "repo_namespace": build.pr.project.namespace,
+                        "web_url": build.web_url,
+                    }
+                    # same_buildid_builds are copr builds created due to the same trigger
+                    # multiple identical builds are created which differ only in target
+                    # so we merge them into one
+                    same_buildid_builds = session.query(CoprBuild).filter(
+                        CoprBuild.build_id == build.build_id
```

Comment:
I think this should be `.filter(build_id=build.build_id)`

Reply:
I'll check if that works, but _CoprBuild.build_id_ is working too.

Reply:
It can't work, because this expression is evaluated before the filter method gets executed so you're passing True/False to the filter function

Reply:
Oh. Then it probably worked for me because I only had three builds (same build_id, different targets)  
And then it combined all of them.